tests/TestCompiler.hs

```diff
@@ -91,11 +91,14 @@ warningTests supportModules supportExterns supportForeigns = do
     wTc <- testSpecs $
       it ("'" <> takeFileName mainPath <> "' should compile with warning(s) '" <> intercalate "', '" expectedWarnings <> "'") $
         assertCompilesWithWarnings supportModules supportExterns supportForeigns testPurs expectedWarnings
-    return $ wTc ++ [ goldenVsString
+    let gwTc = if pathSeparator == '/'
```

Comment:
Does this essentially mean that a developer on windows can't ever run golden tests?

Reply:
Maybe instead of disabling them entirely we can run some sort of normalization/sanitization pass on the output?

Reply:
It does, and we could. It might not be worth the complexity, though? From a do-no-harm perspective, having golden tests that never run on Windows is no worse off than not having the golden tests at all, and it has to be better than having tests that always fail on Windows, or that could cause a Windows developer to check in test outputs that would fail on other platforms.

If we want, probably the best approach would be to use a regex that matches fairly specifically on ‘error found at’ paths (`^at [A-Za-z0-9_.]+(\\[A-Za-z0-9_.]+)+:\d+:\d+`?) and replace all the backslashes inside of them. Worth it?

Reply:
It's not so much about them running in CI, it's that a windows contributor can't even run them to fix them. I think it would be worth it.